src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -181,12 +186,19 @@ private void ValidateOptions(ExecutionOptions options)
                 IValidationResult validationResult;
                 using (metrics.Subject("document", "Validating document"))
                 {
-                    validationResult = _documentValidator.Validate(
-                        config.Query,
-                        config.Schema,
-                        document,
-                        config.ValidationRules,
-                        config.UserContext);
+                    if (config.EnableDocumentValidation)
```

Comment:
I'm really not liking this flag.  I understand why you want to "disable" it, though I'm thinking of too many instances where this would cause problems.  I don't think this should be a feature flag but instead have this behavior in a class that does that for you (which I believe you had in your original post).  We can pass the validator more data if it needs it to properly "disable" validation.

Reply:
@joemcbride 

If I'm understanding right you would prefer something like:
var exec = new DocumentExecutor(new GraphQLDocumentBuilder(), new SkipDocumentValidator(), new ComplexityAnalyzer());

I liked the flag idea since a SkipDocumentValidator is not part of this project.

Note this change alone was responsible for most of the 600% improvements with the starwars query. Most of the time executing small queries is in the validation and would be a huge potential savings in a production environment.

I am very curious on your thoughts. Everything I threw at it seemed to handle appropriately and just give back a more cryptic error since it wouldn't be able to parse the query. It does make me nervous as well, what kind of problems would we need to prevent without this validation?

Reply:
>If I'm understanding right you would prefer something like:
var exec = new DocumentExecutor(new GraphQLDocumentBuilder(), new SkipDocumentValidator(), new ComplexityAnalyzer());

Correct.  This turns off the [Validation feature](http://facebook.github.io/graphql/October2016/#sec-Validation) which is part of the spec.  So everything that Validation does doesn't happen.  Basically, I really don't want to support turning off that feature this easily since that is a pretty big portion of the spec being turned off.  I can understand why you want to do it, though completely turning it off is specific to your business needs and I wouldn't suggest anyone to do it unless they really understand what it means.  Meaning, it would now be easy to write invalid queries and get hard-to-understand errors or null data.

Writing a class like `SkipDocumentValidation` is very easy to do, accomplishes the same thing, and doesn't put the support burden on the core project.

Reply:
Thanks @joemcbride . I'm understanding your point now. I still like it being easier since in production everyone should care about response times but like you mentioned if developers understand the issue they can still implement a custom validator. Maybe as a future update the default validator should be evaluated from a performance standpoint and/or implement a caching strategy.

Removing this features takes the StarWarsPerformanceTest speed improvement from 642% speed gains to 15%.

For our business needs we will plan on removing it completely, since our severs handle a very high amount of traffic.

I've updated my branch. I think I have all of the changes ready that you suggested. Let me know if I'm missing anything.

Reply:
> Maybe as a future update the default validator should be evaluated from a performance standpoint and/or implement a caching strategy.

I think that could be beneficial.  What I've heard from Facebook is they do cache quite a bit.  They have a set of "trusted" queries which I believe would skip validation (since they know it is valid) and even `Document` parsing, since they cache the parsing of the `Document`.

I believe they even go as far as only having the client send a "query name", not even a GraphQL query, and associated variables to cut down on the http traffic.  I added the basic idea of this [here](https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL.GraphiQL/Controllers/GraphQLController.cs#L31-L34) though it doesn't add the `Document` parsing/validation benefit.

Reply:
Cool, that is very good to know. I was playing around with the idea of named queries. Good to know that is the direction Facebook went as well.